pkg/controller/common/layered_node_state.go

```diff
@@ -57,26 +71,26 @@ func (l *LayeredNodeState) isDesiredMachineConfigEqualToPool(mcp *mcfgv1.Machine
 
 // Determines if the nodes desired image is equal to the expected value from
 // the MachineConfigPool.
-func (l *LayeredNodeState) isDesiredImageEqualToPool(mcp *mcfgv1.MachineConfigPool) bool {
-	return l.isImageAnnotationEqualToPool(daemonconsts.DesiredImageAnnotationKey, mcp)
+func (l *LayeredNodeState) isDesiredImageEqualToPool(mcp *mcfgv1.MachineConfigPool, layered bool) bool {
+	return l.isImageAnnotationEqualToPool(daemonconsts.DesiredImageAnnotationKey, mcp, layered)
 }
 
 // Determines if the nodes current image is equal to the expected value from
 // the MachineConfigPool.
-func (l *LayeredNodeState) isCurrentImageEqualToPool(mcp *mcfgv1.MachineConfigPool) bool {
-	return l.isImageAnnotationEqualToPool(daemonconsts.CurrentImageAnnotationKey, mcp)
+func (l *LayeredNodeState) isCurrentImageEqualToPool(mcp *mcfgv1.MachineConfigPool, layered bool) bool {
+	return l.isImageAnnotationEqualToPool(daemonconsts.CurrentImageAnnotationKey, mcp, layered)
 }
 
 // Determines if a nodes' image annotation is equal to the expected value from
 // the MachineConfigPool. If the pool is layered, this value should equal the
 // OS image value, if the value is available. If the pool is not layered, then
 // any image annotations should not be present on the node.
-func (l *LayeredNodeState) isImageAnnotationEqualToPool(anno string, mcp *mcfgv1.MachineConfigPool) bool {
+func (l *LayeredNodeState) isImageAnnotationEqualToPool(anno string, mcp *mcfgv1.MachineConfigPool, layered bool) bool {
 	lps := NewLayeredPoolState(mcp)
 
 	val, ok := l.node.Annotations[anno]
 
-	if lps.IsLayered() && lps.HasOSImage() {
+	if lps.HasOSImage() {
```

Comment:
note: On the surface, it makes sense that if a pool has an OS image that it is layered. But there are two corner-cases that checking for both the presence of the layered label (`IsLayered()`) and the OS image annotation presence (`HasOSImage()`) catches:

1. When a pool is opted into layering, it won't immediately have an OS image because of the time it takes BuildController to build the image and apply it to the pool. In this situation, the node image annotation will not be equal to the pool, so this check should return false.
2. When a pool is opted out of layering, the OS image will be removed from the pool. In this situation, the node will have the image annotation when the pool does not have an OS image, so this check should return false.